### PR TITLE
Fixes Data type mismatch issue on LeaderBoard

### PR DESCRIPTION
**Pull request by padmano**
#302

### CODE REVIEW

**Comment by padmano** [comment]
@Brutus5000 Can you help me out with this ?

**Comment by Brutus5000** [comment]
@iamTheL If you rebase on develop it will be fixed.
That's a nasty circular dependency in our release process.

File: src/main/java/com/faforever/api/leaderboard/LeaderboardService.java [hunk 1]
```diff
@@ -35,8 +37,10 @@ public LeaderboardService(GlobalLeaderboardRepository globalLeaderboardRepositor
     return globalLeaderboardRepository.getLeaderboardByPage(getPageable(page, pageSize));
   }
 
-  public GlobalLeaderboardEntry getGlobalEntry(int playerId) {
-    return globalLeaderboardRepository.findByPlayerId(playerId);
+  public GlobalLeaderboardEntry getGlobalEntry(String playerId) {
+    if(DataTypeValidation.isNumeric(playerId))
+    return globalLeaderboardRepository.findByPlayerId(Integer.valueOf(playerId));
+    else throw new ValidationException();
```
Comment (norraxx) [inline src/main/java/com/faforever/api/leaderboard/LeaderboardService.java]: Brackets?

File: src/main/java/com/faforever/api/leaderboard/LeaderboardService.java [hunk 2]
```diff
@@ -35,8 +37,10 @@ public LeaderboardService(GlobalLeaderboardRepository globalLeaderboardRepositor
     return globalLeaderboardRepository.getLeaderboardByPage(getPageable(page, pageSize));
   }
 
-  public GlobalLeaderboardEntry getGlobalEntry(int playerId) {
-    return globalLeaderboardRepository.findByPlayerId(playerId);
+  public GlobalLeaderboardEntry getGlobalEntry(String playerId) {
+    if(DataTypeValidation.isNumeric(playerId))
```
Comment (1-alex98) [inline src/main/java/com/faforever/api/leaderboard/LeaderboardService.java]: Maybe with Brackets. I actually just read your guidelines and I can not find anything on that but we like to keep code style alike.

Comment (padmano) [reply src/main/java/com/faforever/api/leaderboard/LeaderboardService.java]: Have added the brackets for the `if-else` block. Still it seems to be failing.

**Review by 1-alex98 — DISMISSED** [review]

File: src/main/java/com/faforever/api/leaderboard/LeaderboardService.java (hunk 2 above)
Comment (1-alex98) [reply src/main/java/com/faforever/api/leaderboard/LeaderboardService.java]: What is failing?

File: src/main/java/com/faforever/api/leaderboard/LeaderboardService.java [hunk 3]
```diff
@@ -38,9 +38,11 @@ public LeaderboardService(GlobalLeaderboardRepository globalLeaderboardRepositor
   }
 
   public GlobalLeaderboardEntry getGlobalEntry(String playerId) {
-    if(DataTypeValidation.isNumeric(playerId))
-    return globalLeaderboardRepository.findByPlayerId(Integer.valueOf(playerId));
-    else throw new ValidationException();
+    if (DataTypeValidation.isNumeric(playerId)) {
```
Comment (padmano) [inline src/main/java/com/faforever/api/leaderboard/LeaderboardService.java]: I have added the brackets. Am I missing something else ?

File: src/main/java/com/faforever/api/leaderboard/LeaderboardService.java (hunk 1 above)
Comment (padmano) [reply src/main/java/com/faforever/api/leaderboard/LeaderboardService.java]: I added the brackets too.

File: src/main/java/com/faforever/api/leaderboard/LeaderboardService.java (hunk 2 above)
Comment (padmano) [reply src/main/java/com/faforever/api/leaderboard/LeaderboardService.java]: I am not able to find out what is causing the test coverage to drop.

**Comment by 1-alex98** [comment]
your suggested fix does not work.(tested it)

This should also be way more compact. 
```
if(!StringUtils.isNumeric(playerId)){
      throw new ApiException(new Error(ErrorCode.PLAYER_ID_MUST_BE_NUMBER));
}
```
And use StringUtils from Apache that is it basically.

**Comment by padmano** [comment]
@axel1200 I went ahead with your suggestion.

**Review by 1-alex98 — DISMISSED** [review]

**Review by 1-alex98 — CHANGES_REQUESTED** [review]
I am sorry to annoy you further but this would be a something that can be easily tested, either unit or integration test. Does not really matter.

File: src/main/java/com/faforever/api/leaderboard/LeaderboardService.java (hunk 2 above)
Comment (1-alex98) [reply src/main/java/com/faforever/api/leaderboard/LeaderboardService.java]: Well if u add more lines without writing tests the coverage increases so if u write new stuff(like u did) test it. Make a unit test feed the Service a string and see if it throws the Exception u expected...

Comment (1-alex98) [reply src/main/java/com/faforever/api/leaderboard/LeaderboardService.java]: https://reqtest.com/testing-blog/test-coverage-metrics/

Comment (padmano) [reply src/main/java/com/faforever/api/leaderboard/LeaderboardService.java]: Got it. Will write tests and submit again.

**Comment by Brutus5000** [comment]
Got superseeded by #372 as we replaced the whole exception.

**Pull request closed without merging**